File: mario/less/mario.c
// program that prints out a half-pyramid of a specified height
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    int spaces;
    int hashes;
    int line;
// prompt the user for the half-pyramid’s height, a positive integer no greater than 23
    do
    {
        height = get_int("Height: ");
    }

    while (height < 0 || height > 23);

    for (line = 0; line < height; line++)
    {
// spaces on first line are = to total height - 1
// then keep subtracting 1 space for each next line
        for (spaces = 1; spaces <= height - 1 - line; spaces++)
        {
            printf(" ");
        }

// hashes on first line
        for (hashes = 0; hashes < line + 2; hashes++)
        {
            printf("#");
        }

// start a new line
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
# program that recreates a half-pyramid using hashes (#) for blocks
from cs50 import get_int


def main():
    # prompt and validate user input
    while True:
        height = get_int("Enter a height from 1 to 8, inclusive: ")
        if (height >= 1 and height <= 8):
            break

    # draw the half-pyramid
    for lines in range(0, height):

        # draw the spaces
        for spaces in range(1, height - lines):
            print(" ", end="")

        # draw the hashes
        for hashes in range(lines + 1):
            print("#", end="")

        # print new line
        print()


if __name__ == "__main__":
    main()
